nexus/pack.cc

```diff
@@ -2075,6 +2075,239 @@ struct NexusPacker
         }
     }
 
+    // Finds and returns a flip-flop that drives the given port of an IOB cell
+    // If an associated IOLOGIC cell is provided then checks whether the
+    // flip-flop matches its clock and reset.
+    CellInfo* get_ff_for_iob (CellInfo* iob, IdString port, CellInfo* iol) {
+
+        // Get the net
+        NetInfo* net = get_net_or_empty(iob, port);
+        if (net == nullptr) {
+            return nullptr;
+        }
+
+        // Get the flip-flop that drives it
+        CellInfo* ff = net->driver.cell;
+        if (ff->type != id_OXIDE_FF) {
+            return nullptr;
+        }
+
+        // Get clock nets of IOLOGIC and the flip-flop
```

Comment:
I think this is going to need checks for LSR compatability, and I'm also not sure about how CE works here. Safest is to gate merging them on the `syn_useioff` attribute like ECP5 so it only happens when you explicitly want it and not otherwise.

Reply:
Alright, I can add an attribute controlling that. I couldn't find any mention of such an attribute used by the Vendor tools for crosslink-nx. 

And as I wrote in the description: with this PR It merges only a flip-flop driving T and only if there is an active OLIGIC there. I've noticed the behavior only when using Synplify as the synthesis engine. It didn't happen with LSE.

Reply:
@gatecat I added the requirement for `syn_useioff` to be set to `1` on a net driving `T` for the flip-flop integration to be considered. I also added a check whether the flip flop can be removed.